Code/Editor/Plugins/ComponentEntityEditorPlugin/SandboxIntegration.cpp

```diff
@@ -779,6 +779,7 @@ void SandboxIntegrationManager::OpenPinnedInspector(const AzToolsFramework::Enti
         if (editor && editor->GetPropertyEditor())
         {
             editor->GetPropertyEditor()->SetOverrideEntityIds(entities);
+            editor->SetPinnedInspector();
```

Comment:
We can already deduce if the inspector is pinned or not by checking if `m_overrideSelectedEntityIds` is empty or not. So we don't need to add a new `m_isInspectorPinned` variable to track.

To make it more readable though you could add a new method like `bool InspectorIsPinned() const` that you could use and replace the handful of instances in `EntityPropertyEditor` that are currently checking `m_overrideSelectedEntityIds.empty()`

Reply:
Great work guys. This is a big deal and I'm loving this. I imagine that when we start building out all the asset settings. We might also want to add the pin into the other side of the inspector. Then you can "pin" any kind of inspector. But just some long term thinking

Reply:
It turns out the `EntityPropertyEditor` already has a method called `IsLockedToSpecificEntities()` that returns true if `m_overrideSelectedEntityIds` is not empty, so I replaced it with that